assets/js/components/Dashboard/NewPaMessage/AssociateAlertPage.tsx

```diff
@@ -0,0 +1,349 @@
+import React, {
+  Dispatch,
+  SetStateAction,
+  ComponentType,
+  useState,
+  useEffect,
+} from "react";
+import {
+  Button,
+  Container,
+  Row,
+  Col,
+  ButtonGroup,
+  Modal,
+  Form,
+} from "react-bootstrap";
+import { fetchActiveAndFutureAlerts } from "Utils/api";
+import { Alert } from "Models/alert";
+import classNames from "classnames";
+import { getAlertEarliestStartLatestEnd } from "../../../util";
+import { Page } from "./types";
+import moment from "moment";
+
+interface AssociateAlertPageProps {
+  associatedAlert: Alert;
+  endWithEffectPeriod: boolean;
+  importLocations: boolean;
+  importMessage: boolean;
+  navigateTo: (page: Page) => void;
+  setAssociatedAlert: Dispatch<SetStateAction<Alert>>;
+  setEndWithEffectPeriod: Dispatch<SetStateAction<boolean>>;
+  setImportLocations: Dispatch<SetStateAction<boolean>>;
+  setImportMessage: Dispatch<SetStateAction<boolean>>;
+  setVisualText: Dispatch<SetStateAction<string>>;
+}
+
+const AssociateAlertPage = ({
+  associatedAlert,
+  endWithEffectPeriod,
+  importLocations,
+  importMessage,
```

Comment:
Hmm but don't these values need to be accessible in the parent since that's where the message and locations are defined too? If the values are local state variables in the `AssociateAlertPage` component, is there a way they can be accessed from or passed to the parent without using global state?

Reply:
Maybe these can be callbacks instead? So the parent will still be responsible for handling what happens when something is imported but won't have to keep track of whether an import happened. It makes sense that `endWithAffectPeriod` stays in the parent because it has a component that depends on that value, but these aren't needed once you navigate away. You would still have two props to pass but they could be something like:

```ts
onImportMessage={(alertMessage) => setVisualText(alertMessage)}
onImportLocations={() => ...}
```

But since importing locations is out of scope here, you could just leave it out for now and it can be added when zone selection is ready to add it.

Reply:
Oh interesting. It feels somewhat odd to me to have the flag and callback defined in two different places, but I do see how this simplifies things a bit!

Reply:
Ah but if we do things this way, I think that means we wouldn't be able to conditionally clear things out on "clear" since only the child would know whether or not a message/locations were imported.

Reply:
Yeah that's a good point. Maybe the simple solution is to only clear the associated alert and date import? Leave message and locations how they are and the user can do what they need to to get those updated. If they were to associate with a new alert and import those fields, they would be overwritten anyway.

Reply:
Sounds reasonable to me. Just pushed up the latest - let me know what you think